app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt

```diff
@@ -23,14 +24,16 @@ import org.fossasia.susi.ai.chat.contract.IChatPresenter
 class STTfragment : Fragment(){
     lateinit var recognizer: SpeechRecognizer
     lateinit var chatPresenter: IChatPresenter
+    val activity = ChatActivity()
```

Comment:
Why instantiating a new activity

Reply:
@iamareebjamal There were errors that said 'Smart cast to 'ChatActivity' is impossible, because 'activity' is a property that has open or custom getter'  not possible. So on going through some messages and answers on StackOverflow I realised that this was happening because the type was being inferred each time. The value of the variable could have changed while executing the next statement. Hence, the compiler threw these errors. What I wanted to do is to define the 'activity' once and use it wherever needed rather than leaving it on the compiler to infer the type of 'activity'.

Reply:
This is new activity you are creating. You should use the activity proprty of the class.

Reply:
Yes @iamareebjamal That's right.